### PR TITLE
More timely block import notifications

### PR DESCRIPTION
**Pull request by gavofyork**

### CODE REVIEW

File: substrate/telemetry/src/lib.rs [hunk 1]
```diff
@@ -58,7 +58,7 @@ pub fn init_telemetry(config: TelemetryConfig) -> slog_scope::GlobalLoggerGuard
 					first_time: true,	// ensures that on_connect will be called.
 				}
 			).fuse()
-		).build().fuse(), o!()
+		).chan_size(262144).overflow_strategy(slog_async::OverflowStrategy::DropAndReport).build().fuse(), o!()
```
Comment (pepyakin) [inline substrate/telemetry/src/lib.rs]: What does this number mean?

File: substrate/client/src/client.rs [hunk 2]
```diff
@@ -335,6 +335,7 @@ impl<B, E, Block> Client<B, E, Block> where
 
 		let is_new_best = header.number() == &(self.backend.blockchain().info()?.best_number + One::one());
 		trace!("Imported {}, (#{}), best={}, origin={:?}", hash, header.number(), is_new_best, origin);
+		telemetry!("block.import"; "height" => { let n: u64 = header.number().as_(); n }, "best" => ?hash, "is_new_best" => is_new_best, "origin" => ?origin);
```
Comment (gnunicorn) [inline substrate/client/src/client.rs]: could we please wrap this 158char line ;) ?

File: substrate/telemetry/src/lib.rs (hunk 1 above)
Comment (gnunicorn) [reply substrate/telemetry/src/lib.rs]: 256kb of buffer length to store info in before dropping entries, I suppose, @pepyakin ?

Comment (pepyakin) [reply substrate/telemetry/src/lib.rs]: Yeah, sure. Can we leave a comment? Just a lil bit wary about magic numbers.

File: substrate/client/src/client.rs (hunk 2 above)
Comment (arkpar) [reply substrate/client/src/client.rs]: This should be called after `import_lock` is released in `import_block`.

File: substrate/client/src/client.rs [hunk 3]
```diff
@@ -300,6 +300,11 @@ impl<B, E, Block> Client<B, E, Block> where
 		*self.importing_block.write() = Some(hash);
 		let result = self.execute_and_import_block(origin, hash, header, justification, body);
 		*self.importing_block.write() = None;
+		telemetry!("block.import";
+			"height" => { let n: u64 = header.number().as_(); n },
```
Comment (tomusdrw) [inline substrate/client/src/client.rs]: shouldn't:
```
"height" => header.number().as_::<u64>(),
```
work?

Comment (gavofyork) [reply substrate/client/src/client.rs]: no - `as_` doesn't take a generic param: the trait does.

**Pull request merged**